CoreMapView/src/au/gov/asd/tac/constellation/views/mapview2/MapViewPane.java

```diff
@@ -269,40 +262,36 @@ public void onChanged(final ListChangeListener.Change<? extends String> c) {
-                    if (markerDropDown.getCheckModel().isChecked(item)) {
-                        if (item.equals(MARKER_TYPE_POINT)) {
-                            mapView.updateShowingMarkers(AbstractMarker.MarkerType.POINT_MARKER, true);
-                        } else if (item.equals(MARKER_TYPE_LINE)) {
-                            mapView.updateShowingMarkers(AbstractMarker.MarkerType.LINE_MARKER, true);
-                        } else if (item.equals(MARKER_TYPE_POLYGON)) {
-                            mapView.updateShowingMarkers(AbstractMarker.MarkerType.POLYGON_MARKER, true);
-                        } else if (item.equals(MARKER_TYPE_CLUSTER)) {
-                            mapView.updateShowingMarkers(AbstractMarker.MarkerType.CLUSTER_MARKER, true);
-                        } else if (item.equals(SELECTED_ONLY)) {
-                            mapView.updateShowingMarkers(AbstractMarker.MarkerType.SELECTED, true);
-                        }
-                    } else {
-                        if (item.equals(MARKER_TYPE_POINT)) {
-                            mapView.updateShowingMarkers(AbstractMarker.MarkerType.POINT_MARKER, false);
-                        } else if (item.equals(MARKER_TYPE_LINE)) {
-                            mapView.updateShowingMarkers(AbstractMarker.MarkerType.LINE_MARKER, false);
-                        } else if (item.equals(MARKER_TYPE_POLYGON)) {
-                            mapView.updateShowingMarkers(AbstractMarker.MarkerType.POLYGON_MARKER, false);
-                        } else if (item.equals(MARKER_TYPE_CLUSTER)) {
-                            mapView.updateShowingMarkers(AbstractMarker.MarkerType.CLUSTER_MARKER, false);
-                        } else if (item.equals(SELECTED_ONLY)) {
-                            mapView.updateShowingMarkers(AbstractMarker.MarkerType.SELECTED, false);
-                        }
-
+        markerDropDown.getCheckModel().getCheckedItems().addListener((final ListChangeListener.Change<? extends String> c) -> {
+            markerDropDown.getItems().forEach(item -> {
+                if (markerDropDown.getCheckModel().isChecked(item)) {
+                    if (item.equals(MARKER_TYPE_POINT)) {
+                        mapView.updateShowingMarkers(AbstractMarker.MarkerType.POINT_MARKER, true);
+                    } else if (item.equals(MARKER_TYPE_LINE)) {
+                        mapView.updateShowingMarkers(AbstractMarker.MarkerType.LINE_MARKER, true);
+                    } else if (item.equals(MARKER_TYPE_POLYGON)) {
+                        mapView.updateShowingMarkers(AbstractMarker.MarkerType.POLYGON_MARKER, true);
+                    } else if (item.equals(MARKER_TYPE_CLUSTER)) {
+                        mapView.updateShowingMarkers(AbstractMarker.MarkerType.CLUSTER_MARKER, true);
+                    } else if (item.equals(SELECTED_ONLY)) {
+                        mapView.updateShowingMarkers(AbstractMarker.MarkerType.SELECTED, true);
```

Comment:
just noting that the code in the if and else blocks are exactly the same except passing `true` in the if block and `false` in the else block. Perhaps to reduce code clutter, you could just create a boolean variable for the if condition (the `markerDropDown` one) and then just simply passing the boolean variable into those function calls (removing the need to explicitly call true or false)